File: week3/person.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def __del__(self):
        print(f'{self.name}이(가) 소멸되었습니다.')

    def walk(self):
        print(f'{self.name}이(가) 걷는다.')

    def run(self):
        print(f'{self.name}이(가) 뜁니다.')

p = Person('철수', 20)
p.walk()
p.run()
del p
print('print 함수가 삭제되었습니다.')

File: week3/Person.kt
class Person(private val name: String, private val age: Int) {

    init {
        println("$name object has been created.")
    }

    fun walk() {
        println("$name is walking.")
    }

    fun run() {
        println("$name is running.")
    }

    protected fun finalize() {
        println("$name has been destroyed.")
    }
}

fun main() {
    var p: Person? = Person("Chulsoo", 20)
    p?.walk()
    p?.run()
    p = null
    System.gc() // Force Garbage Collection
    println("The print function has been deleted.")
}